etcd/etcd_test.go

```diff
@@ -116,6 +119,98 @@ func TestReceivesError(t *testing.T) {
 	}
 }
 
+func TestAllEndpointsFailed(t *testing.T) {
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	}))
+
+	s.Close()
+
+	c, err := New(Options{Endpoints: []string{s.URL, s.URL}, Prefix: "/skippertest-invalid"})
+
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	_, err = c.LoadAll()
+
+	if err == nil {
+		t.Error("failed to fail")
+	}
+
+	_, ok := err.(*endpointErrors)
+
+	if !ok {
+		t.Error("when all endpoints fail - endpointErrors must be returned")
+	}
+}
+
+func TestFailedEndpointsRotation(t *testing.T) {
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	}))
+
+	s.Close()
+
+	s2 := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	}))
+
+	c, err := New(Options{Endpoints: []string{s.URL, s2.URL, "neverreached"}, Prefix: "/skippertest-invalid"})
+
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	_, err = c.LoadAll()
+
+	if err == nil {
```

Comment:
i guess this error check is irrelevant here because it doesn't fail this time due to transer error, but because of the empty paylod. fine, though :+1: